client/lib/react-query-devtools-helper/index.jsx

```diff
@@ -0,0 +1,55 @@
+import config from '@automattic/calypso-config';
+import { useEffect, useState } from 'react';
+import ReactDom from 'react-dom';
+import { ReactQueryDevtools } from 'react-query/devtools';
+import { setStoredItem, getStoredItem } from 'calypso/lib/browser-storage';
+
+import './style.scss';
+
+function useDevtoolsEnabled() {
+	const [ enableDevtools, setEnableDevtools ] = useState();
+
+	useEffect( () => {
+		getStoredItem( 'enable-react-query-devtools' ).then( setEnableDevtools );
+	}, [] );
+
+	return enableDevtools;
+}
+
+export function CalypsoReactQueryDevtools() {
+	const enableDevtools = useDevtoolsEnabled();
+
+	if ( config.isEnabled( 'dev/react-query-devtools' ) ) {
+		return enableDevtools ? <ReactQueryDevtools /> : null;
+	}
+
+	return null;
+}
+
+function ReactQueryDevtoolsHelper() {
+	const enableDevtools = useDevtoolsEnabled();
+
+	async function toggleDevtools( event ) {
+		await setStoredItem( 'enable-react-query-devtools', event.target.checked );
+		window.location.reload();
+	}
+
+	return (
+		<>
+			<div>ReactQuery</div>
+			<div className="react-query-devtools-helper__popover">
+				<label className="react-query-devtools-helper__label">
+					<input
+						type="checkbox"
+						name="react-query-devtools"
+						checked={ enableDevtools ?? false }
```

Comment:
The `?? false` could be avoided if `useDevtoolsEnabled()` reliably returned a boolean value. Now, because it uses `useState()` instead of `useState( false )`, it can also return `undefined`.